android/src/main/java/com/amplitude/android/utilities/AndroidStorage.kt

```diff
@@ -37,6 +39,12 @@ class AndroidStorage(
 
     override suspend fun writeEvent(event: BaseEvent) {
         eventsFile.storeEvent(JSONUtil.eventToString(event))
+        val amplitudeExtra = event.extra?.get("amplitude")
+        if (amplitudeExtra != null) {
+            try {
+                event.ingestionMetadata = IngestionMetadata.fromJSONObject((amplitudeExtra as Map<String, Any>).get("ingestionMetadata") as JSONObject)
+            } finally { }
```

Comment:
should this be a `catch`?

Reply:
Why add this in this file? This could be at a general place other than here, also I believe the info would be lost if we do it here.

Reply:
@justin-fiedler No catch here, because, I think any of the exceptions we should ignore set the ingestionMetadata. Maybe it will be better to have some logged info that says, setting ingestionMetadata error? 

@qingzhuozhen I think this is the place before we store in the storage? Why will this get lost?

Reply:
@qingzhuozhen  Which place do you prefer to have the logic? Do you prefer add this logic before writing into the file storage or before sending it to the server?

Reply:
Should this logic be in the AmplitudeDestination plugin rather than storage?